### PR TITLE
Update and clarify readme

### PR DESCRIPTION
**Pull request by PeridexisErrant**
Fixed typos.  Made hyperlinks inline.  Some editorial changes for
clarification.  Updated and expanded introduction.  Fixed a few
formatting errors.

### CODE REVIEW

File: Readme.rst [hunk 1]
```diff
@@ -6,39 +6,56 @@ DFHack Readme
...
 
 ==============
 Getting DFHack
 ==============
-The project is currently hosted on github_
-at http://github.com/DFHack/dfhack
+The project is currently hosted at http://www.github.com/
+
+Recent releases are available in source and binary formats `on the releases
+page`_, while the binaries for releases 40.15-r1 to 34.11-r4 are on DFFD_.
```
Comment (lethosor) [inline Readme.rst]: Could you use the full version numbers here (and below)?

**Comment by lethosor** [comment]
Looks good! (I'm pretty sure that inline links are the ``` `title <http://url/>`_``` form, but I do agree that this way is more readable.)

**Comment by PeridexisErrant** [comment]
> Fix DF version numbers.  Make newlines at headings consistent.  Consistent 4-space indentation where practical.  Fix indented blocks.  Remove duplicate sections.  Minor editorial fixes.

And maybe inline isn't exactly the right phrase - hyperlinks are now text with a link, rather than displaying the link verbatim.  Anyway, I've also pointed wiki links at the current URL form, using `cv:` where applicable so they stay up to date.

**Comment by lethosor** [comment]
There seem to be problems generating the HTML version:
```
Updating Readme.html... Readme.rst:1870: (ERROR/3) Unexpected indentation.
Readme.rst:1870: (WARNING/2) Inline literal start-string without end-string.
Readme.rst:1871: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
```

**Comment by lethosor** [comment]
Fixed.

**Pull request merged**